pallets/payments/src/tests.rs

```diff
@@ -535,6 +537,26 @@ fn request_payment() {
 	})
 }
 
+#[test]
+fn next_id_works() {
+	new_test_ext().execute_with(|| {
```

Comment:
If the reason for the boolean parameter is to allow this block to fail without considering a failure, I'm wondering if `execute_with_safe` could be used instead 🤔

Reply:
not exactly, the reason for the boolean is to skip specific balances in the context of one payment test only. So for the payment id test, when i tried the second assertion, the function failed because it was trying to check the balance for the first payment (which was increased by the first execution).

Taking into account this is a test i wouldn't make it over complexify the logic for this helper function.